tests/test_operations.py

```diff
@@ -303,6 +303,14 @@ def test_shell_escape_escapes_backticks():
     eq_(_shell_escape(cmd), "touch test.pid && kill \`cat test.pid\`")
 
 
+def test_shell_escape_nested_doublequotes():
+    """
+    _shell_escape() escapes nested double-quotes
+    """
+    cmd = 'cd "\\"Application\\" Support"'
+    eq_(_shell_escape(cmd), 'cd \\"\\\\\\\"Application\\\\\\\" Support\\"')
```

Comment:
you can use `r'...'` quoted python strings to make this test more readable (though I'm not sure in what version of python they were introduced) e.g.

```
>>> print('cd \\"\\\\\\\"Application\\\\\\\" Support\\"') ; \
... print(r'cd \"\\\"Application\\\" Support\"')
cd \"\\\"Application\\\" Support\"
cd \"\\\"Application\\\" Support\"
```